src/libostree/ostree-repo-pull.c

```diff
@@ -1530,25 +1530,77 @@ _ostree_repo_verify_bindings (const char  *collection_id,
                               GError     **error)
 {
   g_autoptr(GVariant) metadata = g_variant_get_child_value (commit, 0);
+  g_autoptr(GVariant) collection_refs_binding = NULL;
   g_autofree const char **refs = NULL;
+  gboolean need_ref_binding = FALSE;
+
+  collection_refs_binding = g_variant_lookup_value (metadata, OSTREE_COMMIT_META_KEY_COLLECTION_REFS_BINDING,
+                                                    G_VARIANT_TYPE("a(ss)"));
+
+  if (collection_refs_binding)
+    {
+      if (collection_id != NULL || ref_name != NULL)
```

Comment:
I don't like that. In that case it might take the else case even though collection_refs_binding is non-NULL. In practice, since that will only happen if collection_id == NULL it will not affect the behaviour, but it makes things less obvious.

Reply:
:+1: